src/helm/common/key_value_store.py

```diff
@@ -80,3 +80,32 @@ def multi_put(self, pairs: Iterable[Tuple[Dict, Dict]]) -> None:
     def remove(self, key: Dict) -> None:
         del self._sqlite_dict[key]
         self._sqlite_dict.commit()
+
+
+class BlackHoleKeyValueStore(KeyValueStore):
+    """Key value store that discards all data."""
+
+    def __enter__(self) -> "BlackHoleKeyValueStore":
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        pass
+
+    def contains(self, key: Dict) -> bool:
+        return False
+
+    def get(self, key: Dict) -> Optional[Dict]:
+        return None
+
+    def get_all(self) -> Generator[Tuple[Dict, Dict], None, None]:
+        return
```

Comment:
This return shouldn't be there?

Reply:
```python
return
yield
```

is actually correct and typechecks, as explained [here](https://stackoverflow.com/questions/13243766/how-to-define-an-empty-generator-function). `yield` alone fails typechecking. Also, mypy is okay with this version, whereas the alternatives suggested in the link fail mypy typechecking.